metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py

```diff
@@ -152,6 +152,15 @@ class BigqueryDataset:
     snapshots: List[BigqueryTableSnapshot] = field(default_factory=list)
     columns: List[BigqueryColumn] = field(default_factory=list)
 
+    # Some INFORMATION_SCHEMA views are not available for BigLake tables
+    # based on Amazon S3 and Blob Storage data.
+    # https://cloud.google.com/bigquery/docs/omni-introduction#limitations
+    # Omni Locations - https://cloud.google.com/bigquery/docs/omni-introduction#locations
+    def is_biglake_dataset(self) -> bool:
+        return self.location is not None and self.location.lower().startswith(
+            ("aws-", "azure-")
+        )
```

Comment:
let's add a `supports_table_constraints()` helper here that is just `return not self.is_biglake()`. same for profiling

that will make the rest of the code more clear